roles/libvirt_manager/tasks/create_cloud_init_iso.yml

```diff
@@ -0,0 +1,51 @@
+---
+
+- name: "Acquire authorized_keys from hypervisor"
+  ansible.builtin.slurp:
+    src: "{{ ansible_user_dir ~ '/.ssh/authorized_keys' }}"
+  register: _authorized_keys_b64
+
+- name: "Define the default users for the vm"
+  when:
+    - cifmw_libvirt_manager_vm_users is not defined or
+      cifmw_libvirt_manager_vm_users | length == 0
+  vars:
+    _ssh_authorized_keys: |-
+      [
+      {% for ssh_key in (_authorized_keys_b64.content | b64decode).split('\n') | select() %}
+        "{{ ssh_key | trim }}",
+      {% endfor %}
+      ]
+  ansible.builtin.set_fact:
+    _users:
+      - name: root
+        ssh_authorized_keys: "{{ _ssh_authorized_keys }}"
+        lock_passwd: false
+        plain_text_passwd: "{{ vm_data.password | default('fooBar') }}"
+      - name: "{{ vm_data.user | default('zuul') }}"
+        sudo: ALL=(ALL) NOPASSWD:ALL
+        ssh_authorized_keys: "{{ _ssh_authorized_keys }}"
+        lock_passwd: false
+        plain_text_passwd: "{{ vm_data.password | default('fooBar') }}"
+
+- name: "Define the user data"
+  ansible.builtin.set_fact:
+    _user_data:
+      users: "{{ _users | default(cifmw_libvirt_manager_vm_users) }}"
+      hostname: "{{ vm }}"
+      create_hostname_file: true
+      growpart:
+        devices: [/]
+        ignore_growroot_disabled: true
+        mode: growpart
+      resize_rootfs: noblock
+
+- name: "Call the config_drive role"
+  vars:
+    cifmw_config_drive_iso_image: "{{ _iso_path }}"
+    cifmw_config_drive_uuid: "{{ 99999999 | random(seed=vm) | to_uuid | lower }}"
+    cifmw_config_drive_hostname: "{{ vm }}"
+    cifmw_config_drive_networkconfig: "{{ vm_data.networkconfig | default(None) }}"
```

Comment:
If we are allowing to pass networkconfig, it needs to be possible to define separate config for each VM in the group? i.e if I have compute with amount 3 and want to pass static IPs to each of them? It looks like that won't work, i.e we'd end up setting the same IPs on each VM?

Can we change `networkconfig` to be a list? (Or optional - if it is dict apply same to all VMs - if it is list pick the config from the list using the loop index?

Reply:
i should have fixed this @hjensas , PTAL when you can :)

Reply:
Thanks @rebtoor - looks good now!